code/modules/mob/living/simple_animal/hostile/gym_rats.dm

```diff
@@ -55,6 +59,25 @@
 		attacktext = "punches"
 
 /mob/living/simple_animal/hostile/retaliate/gym_rat/UnarmedAttack(var/atom/A)
+	if(istype(A, /obj/machinery/disposal)) // If we click on a disposal unit and 15 seconds have passed, let's look for meat or cheese
+		to_chat(src, text("<span class='notice'>You rummage through the [A].</span>"))
+
+		if((last_disposalscavenge + disposalscavenge_cooldown >= world.time))
+			to_chat(src, text("<span class='warning'>You scavenged too recently. Wait a little longer.</span>")) // If we scavenged too recently, we get told to wait
```

Comment:
This feels kinda OOC. Could be better if rephrased to something like "You just looked here! You won't find anything new for now." or similar.

Reply:
Alright, I'll adjust it.

Reply:
Should be addressed.